File: javascript/series/series.js
class Series {
  constructor(series) {
    this.digits = series.split('').map(d => Number(d));
  }
  slices(num) {
    if (num > this.digits.length) throw new Error('Slice size is too big.');
    return this.digits.reduce((acc, d, i) => {
      return i < this.digits.length - num + 1 ? [...acc, this.digits.slice(i, i + num)] : acc;
    }, [])
  }
}

export default Series;

File: java/series/src/main/java/Series.java
import java.util.List;
import java.util.stream.IntStream;
import java.util.stream.Collectors;

class Series {
  private final String digits;

  Series(String stringOfDigits) {
    this.digits = stringOfDigits;
  }

  public List<String> slices(int n) {
    if (n > this.digits.length()) {
      throw new IllegalArgumentException("Slice size is too big.");
    }
    if (n < 1) {
      throw new IllegalArgumentException("Slice size is too small.");
    }
    return IntStream.rangeClosed(0, this.digits.length() - n)
      .mapToObj(i -> this.digits.substring(i, i + n))
      .collect(Collectors.toList());
  }
}